### PR TITLE
EOS-23221: Replace SSPL Logger with Cortx Logger class

### PR DESCRIPTION
**Pull request by gargpalak048**
## Problem Statement:

SSPL component should use CORTX logger instead of implementing its own logger class

## Solution Design:

Moving to common CORTX Logger class for SSPL component

## Coding:

* [ ] Did you follow coding standard and verified it with flake8? 

* [ ] Did your address all Codacy issues?

## Testing:

* [ ] Unittest updated and all unittests are passing?

* [ ] Sanity/Self tests are updated (if applicable)?

* [x] Manual testing done covering happy path and non-happy path?

## Integration:

* [ ] If there is any interface change, did you communicate it to other components?
* [ ] If there is any interface change, other componenent gate gatekeepers are ready to accept the change?
* [ ] Did you complete deployment test if applicable?

## PR checklist:
* [x] Did you add Jira number to the PR?

* [x] DCO and cla-signed?

### CODE REVIEW

File: low-level/files/opt/seagate/sspl/setup/sspl_cleanup.py [hunk 1]
```diff
@@ -28,8 +28,8 @@
 from framework.base.sspl_constants import (
     SSPL_CONFIG_INDEX, file_store_config_path, global_config_file_path,
     sspl_config_path, PRODUCT_BASE_DIR, SSPL_BASE_DIR, PRODUCT_FAMILY,
-    RSYSLOG_IEM_CONF, RSYSLOG_SSPL_CONF, RSYSLOG_MSB_CONF,
```
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_cleanup.py:31]: so Cortx logger takes care of log rotation already ! Where is its config, where component specific preferences could be set !?

File: low-level/framework/sspl_init [hunk 2]
```diff
@@ -21,10 +21,8 @@ SCRIPT_NAME=$(basename $0)
 source $script_dir/constants.sh
```
Comment (cdeshmukh) [inline low-level/framework/sspl_init:21]: we need to remove this file all together, legacy code staying here and confusing. anyways changes here am not bothered much

File: low-level/sspl_ll_d [hunk 3]
```diff
@@ -780,8 +780,8 @@ if __name__ == "__main__":
     # Initialize logging
     try:
         logging_level = Conf.get(SSPL_CONF, f"{SYSTEM_INFORMATION}>{LOG_LEVEL}", "INFO")
-        init_logging("SSPL-LL", logging_level, SYSLOG_HOST, SYSLOG_PORT)
-
+        logger.init(service_name="sspl", log_path=SSPL_LOG_PATH, level=logging_level)
+        logger.info("Logging has been initialized for sspl")
```
Comment (cdeshmukh) [inline low-level/sspl_ll_d:784]: keep SSPL is capital. atleast in new / touched code we can maintain that

File: low-level/files/opt/seagate/sspl/setup/sspl_cleanup.py (hunk 1 above)
Comment (gargpalak048) [reply low-level/files/opt/seagate/sspl/setup/sspl_cleanup.py:31]: > so Cortx logger takes care of log rotation already
yes

> Where is its config, where component specific preferences could be set !?
These are the default config params,
backup_count=10,
file_size_in_mb=10,

We can parse these params in init method, to change the config values

File: low-level/sspl_ll_d (hunk 3 above)
Comment (gargpalak048) [reply low-level/sspl_ll_d:784]: okay

File: low-level/framework/base/internal_msgQ.py [hunk 4]
```diff
@@ -20,7 +20,7 @@
                     another.
  ****************************************************************************
 """
-from framework.utils.service_logging import logger
+from cortx.utils.log import Log as logger
```
Comment (thavanathan) [inline low-level/framework/base/internal_msgQ.py:23]: logger is a common name used in many projects. We should ask cortx utils to change it to logger.

**Review by thavanathan — COMMENTED** [review]
initial review.

File: low-level/framework/utils/utility.py [hunk 5]
```diff
@@ -152,14 +152,15 @@ def replace_expr(filename:str, key, new_str:str):
                 f.write(line)
 
     @staticmethod
-    def set_ownership_recursively(root_dir, uid, gid):
+    def set_ownership_recursively(list_root_dir, uid, gid):
```
Comment (sandeepsgt) [inline low-level/framework/utils/utility.py:155]: Why this change is required ?

**Review by sandeepsgt — APPROVED** [review]

**Review by sumedhak27 — APPROVED** [review]

File: low-level/actuators/impl/centos_7/command_line.py [hunk 6]
```diff
@@ -24,7 +24,7 @@
 from actuators.Icommand_line import ICommandLine
 
 from framework.base.debug import Debug
-from framework.utils.service_logging import logger
+from cortx.utils.log import Log as logger
```
[the remark this answers is not in the document]
Comment (gargpalak048) [reply low-level/actuators/impl/centos_7/command_line.py:27]: Ack Chetan.
This actually reduced so many efforts & saved time.

**Comment by gargpalak048** [comment]
Raising a new PR: https://github.com/Seagate/cortx-monitor/pull/521
which has better approach & leads to less no of file changes
